photutils/background/core.py

```diff
@@ -0,0 +1,675 @@
+        The number of iterations to perform sigma clipping, or `None` to
+        clip until convergence is achieved (i.e., continue until the
+        last iteration clips nothing). Defaults to 5.
+
+    Examples
+    --------
+    >>> from photutils import MeanBackground
+    >>> data = np.arange(100)
+    >>> bkg = MeanBackground(sigma=3.)
+
+    The background value can be calculated by using the
+    ``.calc_background()`` method, e.g.:
+
+    >>> bkg_value = bkg.calc_background(data)
+    >>> print(bkg_value)    # doctest: +FLOAT_CMP
+    49.5
+
+    Alternatively, the background value can be calculated by calling the
+    class instance as a function, e.g.:
+
+    >>> bkg_value = bkg(data)
+    >>> print(bkg_value)    # doctest: +FLOAT_CMP
+    49.5
+    """
+
+    def __init__(self, **kwargs):
+
+        super(MeanBackground, self).__init__(**kwargs)
+
+    def calc_background(self, data):
+
+        if self.sigclip:
+            data = self.sigma_clip(data)
+        return np.ma.mean(data)
+
+
+class MedianBackground(BackgroundBase):
+    """
+    Class to calculate the background in an array as the (sigma-clipped)
+    median.
+
+    Parameters
+    ----------
+    sigma : float, optional
+        The number of standard deviations to use for both the lower and
+        upper clipping limit. These limits are overridden by
+        ``sigma_lower`` and ``sigma_upper``, if input. Defaults to 3.
+    sigma_lower : float or `None`, optional
+        The number of standard deviations to use as the lower bound for
+        the clipping limit. If `None` then the value of ``sigma`` is
+        used. Defaults to `None`.
+    sigma_upper : float or `None`, optional
+        The number of standard deviations to use as the upper bound for
+        the clipping limit. If `None` then the value of ``sigma`` is
+        used. Defaults to `None`.
+    iters : int or `None`, optional
+        The number of iterations to perform sigma clipping, or `None` to
+        clip until convergence is achieved (i.e., continue until the
+        last iteration clips nothing). Defaults to 5.
+
+    Examples
+    --------
+    >>> from photutils import MedianBackground
+    >>> data = np.arange(100)
+    >>> bkg = MedianBackground(sigma=3.)
+
+    The background value can be calculated by using the
+    ``.calc_background()`` method, e.g.:
+
+    >>> bkg_value = bkg.calc_background(data)
+    >>> print(bkg_value)    # doctest: +FLOAT_CMP
+    49.5
+
+    Alternatively, the background value can be calculated by calling the
+    class instance as a function, e.g.:
+
+    >>> bkg_value = bkg(data)
+    >>> print(bkg_value)    # doctest: +FLOAT_CMP
+    49.5
+    """
+
+    def __init__(self, **kwargs):
+
+        super(MedianBackground, self).__init__(**kwargs)
+
+    def calc_background(self, data):
+
+        if self.sigclip:
+            data = self.sigma_clip(data)
+        return np.ma.median(data)
+
+
+class MMMBackground(BackgroundBase):
```

Comment:
This is almost identical to `SExtractorBackground` - and I know of other people who prefer slightly different factors to go with median and mean.  Maybe adapt this into somthing like `ModeEstimatorBackground` but with the 3/2 as *parameters*?  Then `MMMBackground` and `SExtractorBackground` can be simple classes along the line of:

```python
class MMMBackground(ModeEstimatorBackground):
    def __init__(self, **kwargs):
        kwargs['median_factor'] = 3
        kwargs['mean_factor'] = 2
        super(MMMBackground, self).__init__(**kwargs)

class SExtractorBackground(ModeEstimatorBackground):
    def __init__(self, **kwargs):
        kwargs['median_factor'] = 2.5
        kwargs['mean_factor'] = 1.5
        super(MMMBackground, self).__init__(**kwargs)